overlord/snapstate/snapmgr.go

```diff
@@ -161,6 +162,43 @@ func (snapst *SnapState) CurrentSideInfo() *snap.SideInfo {
 	panic("cannot find snapst.Current in the snapst.Sequence")
 }
 
+func (snapst *SnapState) PreviousSideInfo() *snap.SideInfo {
```

Comment:
should this be private? or have a comment?

should maybe start with a check for HasCurrent as CurrentSideInfo does